recipes/qcoro/all/conanfile.py

```diff
@@ -23,124 +32,149 @@ class QCoroConan(ConanFile):
         "fPIC": True,
         "asan": False,
     }
-    generators = "cmake", "cmake_find_package_multi"
-    exports_sources = ["CMakeLists.txt"]
-    _cmake = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
 
     @property
     def _compilers_minimum_version(self):
-        minimum_versions = {
-                "gcc": "10",
-                "Visual Studio": "17",
-                "msvc": "19.29",
-                "clang": "8",
-                "apple-clang": "13"
+        return {
+            "gcc": "10",
+            "Visual Studio": "17",
+            "msvc": "192",
+            "clang": "8",
+            "apple-clang": "13",
         }
-        return minimum_versions
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
 
-    def build_requirements(self):
-        self.build_requires("cmake/3.23.2")
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("qt/6.3.1")
+        self.requires("qt/[>=6.6.0 <7]", transitive_headers=True, transitive_libs=True)
```

Comment:
I believe that qcoro is also compatible with Qt5 looking at the logic in this file: https://github.com/qcoro/qcoro/blob/v0.10.0/cmake/QCoroFindQt.cmake

Reply:
Sure, but how would you change the recipe accordingly? Add a `"qt": [5, 6]` option? The user is already able to `force=True` an older version, if necessary.

Reply:
I would just change version range by something like this `qt/[>=5.15 <7]`. I don't see any reason to add an option.

Reply:
I can add that, but it probably won't have any real benefit, though. Conan does practically no conflict resolution and will most likely just complain and fail due to the other recipes are not using the newest version that is <7.

Reply:
This would allow for this to be used when users have an extra dependency on QT that is pinned/resolves to a 5.x version.

If the changes to the recipe to support the older version are just allowing it in the version range, I'd go for it with a comment, otherwise if it's more difficult than that, we can keep the current approach